src/cmd-artifact-disk

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python3 -u
+import argparse
+import logging as log
+import os.path
+import shutil
+import pydoc
+import sys
+
+cosa_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, f"{cosa_dir}/cosalib")
+sys.path.insert(0, cosa_dir)
+
+from cosalib.cmdlib import get_basearch
+
+try:
+    from cosalib.build import _Build
+except ImportError:
+    # We're in the container and can't sense the cosa path
+    sys.path.insert(0, '/usr/lib/coreos-assembler/cosalib')
+    from cosalib.build import _Build
+
+IMG_CLASS_MAP = {
+    "aliyun": "cosalib.qemu.AliyunImage",
+    "aws": "cosalib.qemu.AWSImage",
+    "azure": "cosalib.qemu.AzureImage",
```

Comment:
One recent trend in programming languages has been away from object orientation.  While there are a few problem domains it matches really well (GUI programming), I think having good code *organization* (like Rust modules, Go packages) with basic procedures and interface types is generally better - more flexible, doesn't constrain one as much into thinking of "is-a" relationships.

Specifically for us, I think this code is a good example of when object orientation isn't a great fit - we have a list here of known subclasses.

And in general build software seems very much to me a problem domain that's procedural with functional parts.

That doesn't mean we don't create objects - Python clearly needs them, I just I think it'd be more natural to basically use classes as "structs with methods" - no inheritance.  So e.g. the `buildextend-aws` would call `build_image({ platformid: 'aws', 'image_suffix': 'vmdk'})` etc., and that may end up creating an ImageBuild struct with convenient parameters, but there wouldn't be subclassing.

That all said, this is definitely an improvement overall!  So none of this is a blocker, just writing out my thoughts, and we should try to be sure we're aligned on future directions.

Reply:
(Obviously, Python the language is heavily about objects so it's hard to avoid entirely, and doing so wouldn't be worth it!  I just program in Rust and Go (and C) enough that I don't tend to think "subclass" unless the problem domain really really needs it)

Reply:
I appreciate the thought and the discussion, and I agree that having an interface over OOO would be nicer. While hacking on this I had similar thoughts, especially around the `_Build` Class. 

Though, the feedback is valuable. I'm seeing a clear case where the VMWare OVA can become a subclass, but the others are weaker cases, especially when it amounts to settings some  `**kwargs`

Reply:
For anyone who hasn't seen it: http://steve-yegge.blogspot.com/2006/03/execution-in-kingdom-of-nouns.html

Reply:
Good read. Thanks Colin. Java does tend to go overboard in it's OO implementation and expectations.